flask_app/blueprints/rest.py

```diff
@@ -35,6 +36,75 @@ def decorator(resource):
 
 ##########################################################################
 
+session_metadata_parser = reqparse.RequestParser()
+session_metadata_parser.add_argument("id", type=int, default=None)
+session_metadata_parser.add_argument("key", type=str, default=None)
+session_metadata_parser.add_argument("value", type=str, default=None)
+
+@_resource('/session-metadata', '/session-metadata/<object_id>')
+class SessionMetadataResource(ModelResource):
+
+    MODEL = SessionMetadata
+
+    def _get_iterator(self):
+        args = session_metadata_parser.parse_args()
+
+        returned = super(SessionMetadataResource, self)._get_iterator()
+
+        # filter by session id
+        if args.id is not None:
+            returned = returned.filter(SessionMetadata.session_id == args.id)
+
+        # filter by key
+        if args.key is not None:
+            returned = returned.filter(SessionMetadata.key == args.key)
+
+        # filter by value
+        # TODO: Allow generic parsing of jsonb column data (need to add "" around string )
```

Comment:
Is this something you still want todo?

Reply:
Hmm, Im not exactly sure what the best path forward is. Either I restrict the user to use specific query parameters, or the user will need to know the structure of the value as a json object.

Reply:
For report generation purposes this is not necessary.